src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAITextToImageClient.cs

```diff
@@ -0,0 +1,252 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+using OpenAI;
+using OpenAI.Images;
+
+#pragma warning disable S3011 // Reflection should not be used to increase accessibility of classes, methods, or fields
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents an <see cref="ITextToImageClient"/> for an OpenAI <see cref="OpenAIClient"/> or <see cref="ImageClient"/>.</summary>
+internal sealed class OpenAITextToImageClient : ITextToImageClient
+{
+    /// <summary>Metadata about the client.</summary>
+    private readonly TextToImageClientMetadata _metadata;
+
+    /// <summary>The underlying <see cref="ImageClient" />.</summary>
+    private readonly ImageClient _imageClient;
+
+    /// <summary>The underlying <see cref="OpenAIClient" />.</summary>
+    private readonly OpenAIClient? _openAIClient;
```

Comment:
I think we can drop this. We don't store the OpenAIClient in the other modalities. We then just don't support returning it from GetService.